cibyl/sources/zuul/query.py

```diff
@@ -214,44 +279,21 @@ def handle_query(api, **kwargs):
     :return: Attribute assembled for usage by Cibyl.
     :rtype: :class:`AttributeDictValue`
     """
-    model = ModelBuilder()
-    query = get_query_type(**kwargs)
-
-    if query == QueryType.TENANTS:
-        for tenant in _get_tenants(api, **kwargs):
-            model.with_tenant(tenant)
-
-    if query == QueryType.PROJECTS:
-        for project in _get_projects(api, **kwargs):
-            model.with_project(project)
+    handlers = {
+        QueryType.TENANTS: _handle_tenants_query,
+        QueryType.PROJECTS: _handle_projects_query,
+        QueryType.PIPELINES: _handle_pipelines_query,
+        QueryType.JOBS: _handle_jobs_query,
+        QueryType.BUILDS: _handle_jobs_query
+    }
 
-    if query == QueryType.PIPELINES:
-        for pipeline in _get_pipelines(api, **kwargs):
-            model.with_pipeline(pipeline)
-
-    if query == QueryType.JOBS:
-        pipelines = _get_pipelines(api, **kwargs)
-
-        for job in _get_jobs(api, **kwargs):
-            job_model = model.with_job(job)
-
-            # Include also the pipelines where this job is present
-            for pipeline in pipelines:
-                pipeline_jobs = [j.name for j in pipeline.jobs().get()]
-
-                # Is the job on this pipeline?
-                if job.name in pipeline_jobs:
-                    pipeline_model = model.with_pipeline(pipeline)
-                    pipeline_model.add_job(job_model)
+    query = get_query_type(**kwargs)
+    handler = handlers.get(query)
 
-            # Check if the user requested variants
-            if 'variants' in kwargs:
-                for variant in job.variants().get():
-                    job_model.add_variant(Job.Variant.from_data(variant.data))
+    if not handler:
+        raise NotImplementedError(f'Unsupported query: {query}')
```

Comment:
This exception should be CibylNotImplementedException

Reply:
Sure, will do.